checker/checkers.go

```diff
@@ -114,13 +116,25 @@ type typeAndPriority struct {
 }
 
 func determineTypeAndPriorityForPaths(assetPaths []string) (map[string]typeAndPriority, error) {
-	typeAndPriorities := make(map[string]typeAndPriority)
-	for _, assetPath := range assetPaths {
-		typeAndPriorityForAsset, err := determineTypeAndPriority(assetPath)
-		if err != nil {
-			return nil, err
-		}
-		typeAndPriorities[assetPath] = typeAndPriorityForAsset
+	typeAndPriorities := map[string]typeAndPriority{}
+	var mapLock sync.Mutex
+	g := errgroup.Group{}
+	for _, assetPathSingle := range assetPaths {
+		assetPath := assetPathSingle
+		g.Go(func() error {
+			typeAndPriorityForAsset, err := determineTypeAndPriority(assetPath)
+			if err != nil {
+				return err
+			}
+			mapLock.Lock()
+			typeAndPriorities[assetPath] = typeAndPriorityForAsset
+			mapLock.Unlock()
+			return nil
+		})
+	}
+	err := g.Wait()
+	if err != nil {
```

Comment:
nit: inline error assignment and check if other variables are not being stored/updated

```suggestion
	if err:= g.Wait(); err != nil {
```

Reply:
Done